--- plugin.ori.js	2020-02-29 23:27:08.195407500 +0100
+++ plugin.js	2020-02-29 23:19:59.842964500 +0100
@@ -1,5 +1,6 @@
 ﻿/**
  * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ *                    (c) 2019, Inkscape-web, Martin Owens, AGPLv3
  * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
  */
 
@@ -119,7 +120,36 @@
 
 				// Now we have all the blocks to be included in a new blockquote node.
 				var bqBlock = editor.document.createElement( 'blockquote' );
+
+				var quoteBox = undefined;
+				if (paragraphs[0].$.innerText == "\n") {
+						// The quoteBox is a special variable that contains selected text
+						// From anywhere in the forum website.
+						var quoteBox = localStorage.getItem('quoteBox');
+						if (quoteBox) {
+							localStorage.setItem("quoteBox", "");
+							paragraphs[0].$.prepend(quoteBox);
+						}
+				}
 				bqBlock.insertBefore( paragraphs[ 0 ] );
+
+				var quoteAuthor = localStorage.getItem('quoteAuthor');
+				var quoteUrl = localStorage.getItem('quoteUrl');
+				if (quoteAuthor) {
+						bqBlock.setAttribute('data-author', quoteAuthor);
+						var cite = editor.document.createElement('cite');
+						if (quoteUrl) {
+								bqBlock.setAttribute('data-cite', quoteUrl);
+								var anchor = editor.document.createElement('a');
+								anchor.setAttribute('href', quoteUrl);
+								anchor.$.prepend(quoteAuthor);
+								cite.append(anchor);
+						} else {
+								cite.$.prepend(quoteAuthor);
+						}
+						bqBlock.append(cite);
+				}
+
 				while ( paragraphs.length > 0 ) {
 					block = paragraphs.shift();
 					bqBlock.append( block );
